x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/SimpleRole.java

```diff
@@ -168,6 +175,74 @@ public IndicesAccessControl authorize(
         return indices.authorize(action, requestedIndicesOrAliases, aliasAndIndexLookup, fieldPermissionsCache);
     }
 
+    @Override
+    public RoleDescriptorsIntersection getRemoteAccessRoleDescriptorsIntersection(final String remoteClusterAlias) {
+        final RemoteIndicesPermission remoteIndicesPermission = remoteIndices.forCluster(remoteClusterAlias);
+        if (remoteIndicesPermission.remoteIndicesGroups().isEmpty()) {
+            return RoleDescriptorsIntersection.EMPTY;
+        }
+        final List<RoleDescriptor.IndicesPrivileges> indicesPrivileges = new ArrayList<>();
+        for (RemoteIndicesPermission.RemoteIndicesGroup remoteIndicesGroup : remoteIndicesPermission.remoteIndicesGroups()) {
+            for (IndicesPermission.Group indicesGroup : remoteIndicesGroup.indicesPermissionGroups()) {
+                indicesPrivileges.add(toIndicesPrivileges(indicesGroup));
+            }
+        }
+
+        return new RoleDescriptorsIntersection(
+            new RoleDescriptor(
+                REMOTE_USER_ROLE_NAME,
```

Comment:
I've preserved the same approach and kept the same name for **all** remote access role descriptors. As a result to this, we would end up having an intersection which consists of sets containing single role descriptor with the same name `REMOTE_USER_ROLE_NAME`. Meaning, the name is unique within a set (since it contains only one role), but it is not unique within an intersection list.

I've tried running simple remote authentication tests and did not see any issues. But I fear I might be missing some edge cases.

@n1v0lg Of the top of your mind, do you see any obvious issue with having the same name for all role descriptors in the intersection?

Reply:
Good q! I don't think there's an issue here. Even beyond remote cluster security, it's perfectly acceptable to have a user with role name `role-a` create an API key with assigned role name `role-a` -- we don't rely on role descriptor names being unique (different story of course when we create roles with the roles API but that's not applicable here). 

For remote cluster security, the role name is essentially throw away from a fulfilling cluster perspective.